### PR TITLE
Release candidate/0.0.1 rc17

### PR DESCRIPTION
**Pull request by W3stside**
## Changes

1. Uses `@gnosis.pm/dutchx-verification-modal` code to present `Terms`, `Cookies`, and `Privacy` policies
2. linting
3. logic

### CODE REVIEW

**Comment by ghost** [comment]
Travis automatic deployment:
 https://pr10--dxmgnpoolreact.review.gnosisdev.com

**Comment by ghost** [comment]
Travis automatic deployment:
 https://pr10--dxmgnpoolreact.review.gnosisdev.com

File: webpack.config.js [hunk 1]
```diff
@@ -124,6 +124,9 @@ module.exports = (_, { mode }) => {
     },
     resolve: {
       extensions: ['.js', '.jsx', '.json'],
+      alias: {
+        react: path.resolve('./node_modules/react'),
```
Comment (Velenir) [inline webpack.config.js:128]: What is this for? Aliasing is usually done when you want to replace imports of say `react` with `preact`...

File: src/components/controls/WalletIntegration.jsx [hunk 2]
```diff
@@ -58,6 +50,15 @@ function WalletIntegration({
       // Set Modal
       showModal('Loading user data . . .')
 
+      // Gnosis Safe Fix
+      Promise.race([
+        delay(500),
+        safeInjected,
```
Comment (Velenir) [inline src/components/controls/WalletIntegration.jsx:56]: If you want there's a proper `safeInjected` implementation in https://github.com/gnosis/dx-mgn-pool-react/blob/99e490c5e2a6d878b8db4303c62f059b2ecd65c4/src/api/providers/index.js#L78 that waits for an event from Safe on `window`
But delay works too.

File: src/api/utils/index.js [hunk 3]
```diff
@@ -19,6 +19,48 @@ const windowLoaded = new Promise((resolve) => {
...
+    console.warn('Non-Metamask or Gnosis Safe Provider injected web3 API detected')
+
+    window.web3 = web3 = new Web3(web3.currentProvider)
+  }
+
+  // 1.X.X API
+  if (typeof web3.version === 'string') {
+    netID = await new Promise((accept, reject) => {
+      web3.eth.net.getId((err, res) => {
+        if (err) {
```
Comment (Velenir) [inline src/api/utils/index.js:42]: Doesn't `web3.eth.net.getId()` return a Promise by itself if you don't pass in a callback?

**Review by Velenir — APPROVED** [review]
Good work on the modal.

File: webpack.config.js (hunk 1 above)
Comment (W3stside) [reply webpack.config.js:128]: making sure to use the correct react when there could posibly be 2 diff versions - i had an issue with useEffect aka 16.8+ react and another verison. `npm ls react` didnt show me any differences but this was the only solution that worked for me

File: src/api/utils/index.js (hunk 3 above)
Comment (W3stside) [reply src/api/utils/index.js:42]: it does now, i believe it didnt even in 1.0 when i made this, ill change it

File: src/components/controls/WalletIntegration.jsx (hunk 2 above)
Comment (W3stside) [reply src/components/controls/WalletIntegration.jsx:56]: idk this was your code i moved :man_shrugging:

Comment (W3stside) [reply src/components/controls/WalletIntegration.jsx:56]: will take a look at this tho thanks

**Pull request merged**